src/ddspdrum/module.py

```diff
@@ -0,0 +1,239 @@
+"""
+Synth modules.
+
+    TODO    :   - VCA is slow. Blocks? Not sure what's best for tensors.
+                - Convert operations to tensors, obvs.
+"""
+
+import numpy as np
+from scipy.signal import resample
+
+from ddspdrum.defaults import CONTROL_RATE, SAMPLE_RATE
+
+
+class SynthModule:
+    """
+    Base class for synthesis modules. Mostly helper functions for the moment.
+    """
+
+    def __init__(
+        self, sample_rate: int = SAMPLE_RATE, control_rate: int = CONTROL_RATE
+    ):
+        self.sample_rate = SAMPLE_RATE
+        self.control_rate = CONTROL_RATE
+
+    def control_to_sample_rate(self, control: np.array) -> np.array:
+        """
+        Resample a control signal to the sample rate
```

Comment:
@jorshi and @maxsolomonhenry I'd like your thoughts on this suggestion.

Reply:
I've been thinking about the different rates for control signals and audio signals -- I like the idea and I think it makes sense for performance concerns. I'm also wondering if it is creating too much complexity at this point in time, especially considering we are resampling control signals up to audio signal rate when it comes time to apply them to signals. Should everything just run at audio rate?

Reply:
I was wondering that myself. The benefit of a control rate comes when you're composing lots of control signals; LFOs and envelopes and whatnot. But given that this isn't really meant to run in realtime or anything, I see your point. Maybe let's leave it for now? Can always remove it later.

Reply:
@maxsolomonhenry what is the benefit? speed?

I do feel we should stop working in the domain of seconds and instead make nsamples (in sample rate domain) the primary unit of duration, pretty soon.

Reply:
We have to keep the time components articulated in time values... we don't want them scaling with different sample rates.

Reply:
Got it.